### PR TITLE
Algorithms from Images.jl to form base

### PR DESCRIPTION
**Pull request by ashwani-rathee**
- Part of restructuring effort to get corner related algorithms outside of Images.jl and hence ensuring that when accessing corner related algorithms, we don't have to install Images.jl directly which is a heavy dependency. This is needed for Pure Julia ProjectiveGeometry and Camera Calibration solutions. I intend to bring some more algorithms I saw in ImageProjectiveGeometry.jl

### CODE REVIEW

**Comment by timholy** [comment]
Splitting this out is a good idea. Is corner detection similar enough to edge-detection that combining this with https://github.com/JuliaImages/ImageEdgeDetection.jl makes sense? It seems extensive enough that it's also fine to leave it as a standalone function.

**Comment by ashwani-rathee** [comment]
We sure could combine ImageCorners.jl and ImageEdgeDetection.jl given only 3 corner detectors atm but when we bring in algorithms from ImageProjectiveGeometry.jl here: https://github.com/peterkovesi/ImageProjectiveGeometry.jl/blob/master/src/cornerfeatures.jl it will start to get bit cluttered. Also keeping these packages modular and very small allows more breathing space for experimentation for model based methods that I want to work on.

**Comment by zygmuntszpak** [comment]
I recommend keeping ImageCorners separate to ImageEdgeDetection. The non-maxima suppression that is used in edge detection is somewhat different from the kind of non-maxima suppression that is often utilised by many corner detection algorithms.

**Comment by timholy** [comment]
Sounds good to me, thanks for tackling this @ashwani-rathee.

**Comment by ashwani-rathee** [comment]
Couple of questions:
- Any preferences on what colors we definitely want to precompile? I currently just pick what @timholy and @johnnychen94 use.
- Which of the methods in my precompile file are redundant/not needed?
- Why is it when I explicitly tell it to precompile, I don't see the @info string I mentioned in precompile.jl? Is it even precompiling again?

File: Project.toml [hunk 1]
```diff
@@ -2,3 +2,21 @@ name = "ImageCorners"
...
+ImageFiltering = "6a3955dd-da59-5b1f-98d4-e7296123deb5"
+PrecompileTools = "aea7be01-6a6a-4083-8856-8a6e6704d82a"
+StaticArrays = "90137ffa-7385-5640-81b9-e52037218182"
+StatsBase = "2913bbd2-ae8a-5f71-8c99-4fb6c76f3a91"
+
+[compat]
+ColorTypes = "0.11"
+ColorVectorSpace = "0.9"
+ImageCore = "0.9"
+ImageFiltering = "0.7"
```
Comment (timholy) [inline Project.toml:19]: Need to add PrecompileTools

File: src/precompile.jl [hunk 2]
```diff
@@ -0,0 +1,55 @@
+using PrecompileTools
+
+@info "Precompiling ImageCorners"
```
Comment (timholy) [inline src/precompile.jl:3]: Julia issues this information already, not needed

File: src/precompile.jl [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+@setup_workload begin
+
+    imgs_list = Any[
+        rand(Gray{ColorTypes.N0f8}, 32, 32),
+        rand(RGB{ColorTypes.N0f8}, 32, 32),
+        rand(Gray{ColorTypes.Float64}, 32, 32),
+        rand(RGB{Float64}, 32, 32),
+        rand(ColorTypes.N0f8, 32, 32),
+        rand(ColorTypes.Float64, 32, 32)
+    ]
```
Comment (timholy) [inline src/precompile.jl:14]: These colors seem fine.

File: src/ImageCorners.jl [hunk 4]
```diff
@@ -1,5 +1,28 @@
...
+using ColorTypes
+using ColorVectorSpace
+using ImageCore: NumberLike
+using ImageFiltering
+using PrecompileTools
+using StaticArrays
+using StatsBase
+
+include("utils.jl")
+include("cornerapi.jl")
```
Comment (timholy) [inline src/ImageCorners.jl:12]: Add `include("precompile.jl")` if you want it to have any effect.

**Review by timholy — COMMENTED** [review]
Just a couple of items I noticed

**Pull request merged**